src/telegram/_suggestedpost.py

```diff
@@ -154,6 +154,80 @@ def de_json(cls, data: JSONDict, bot: Optional["Bot"] = None) -> "SuggestedPostP
         return super().de_json(data=data, bot=bot)
 
 
+class SuggestedPostInfo(TelegramObject):
+    """
+    Contains information about a suggested post.
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`state` and :attr:`price` are equal.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        state (:obj:`str`):
+            State of the suggested post. Currently, it can be one of
+            :tg-const:`~telegram.constants.SuggestedPostInfoState.PENDING`,
+            :tg-const:`~telegram.constants.SuggestedPostInfoState.APPROVED`,
+            :tg-const:`~telegram.constants.SuggestedPostInfoState.DECLINED`.
+        price (:obj:`SuggestedPostPrice`, optional):
+            Proposed price of the post. If the field is omitted, then the post is unpaid.
+        send_date (:class:`datetime.datetime`, optional):
+            Proposed send date of the post. If the field is omitted, then the post can be published
+            at any time within 30 days at the sole discretion of the user or administrator who
+            approves it.
+            |datetime_localization|
+
+    Attributes:
+        state (:obj:`str`):
+            State of the suggested post. Currently, it can be one of
+            :tg-const:`~telegram.constants.SuggestedPostInfoState.PENDING`,
+            :tg-const:`~telegram.constants.SuggestedPostInfoState.APPROVED`,
+            :tg-const:`~telegram.constants.SuggestedPostInfoState.DECLINED`.
+        price (:obj:`SuggestedPostPrice`):
+            Optional. Proposed price of the post. If the field is omitted, then the post is unpaid.
+        send_date (:class:`datetime.datetime`):
+            Optional. Proposed send date of the post. If the field is omitted, then the post can be
+            published at any time within 30 days at the sole discretion of the user or
+            administrator who approves it.
+            |datetime_localization|
+
+    """
+
+    __slots__ = ("price", "send_date", "state")
+
+    def __init__(
+        self,
+        state: Literal["pending", "approved", "declined"],
+        price: Optional[SuggestedPostPrice] = None,
+        send_date: Optional[dtm.datetime] = None,
+        *,
+        api_kwargs: Optional[JSONDict] = None,
+    ):
+        super().__init__(api_kwargs=api_kwargs)
+        # Required
+        self.state: str = state
```

Comment:
can we convert this to SuggestedPostInfoState? We do something like that for all the `*Type` enums